### ISSUE TITLE
Help needed to understand the code

### ISSUE THREAD

**Issue by scigeek72** [orybkin/sigma-vae-tensorflow#7]
Hi,
I have been trying to do a similar thing as in the paper (of which this github repo is a part). I wanted to learn the 'sigma' along with the reconstruction (mu) of the input. By mu and sigma, I am referring to the likelihood or decoder distribution, i,e, p(x|z) ~ N(mu, sigma**2*I) instead of just considering p(x|z) ~ N(mu, I) like what is being done in the calibrated decoder paper associated with this repo. For me, the goal is to actually calculate the probability of reconstructed output for a given latent z: lower the probability, poorer the reconstruction. For implementation, I looked at your repo and wrote a very informal script in Keras + TF. My model is running. But I noticed the loss is very high (although over the course of 20 epochs, it does going down). So in your code, on line 51, I noticed you divided the loss by a number, ( AdamOptimizer.minimize(loss/np.sum(img.shape).value). Isn't np.sum() gives a scalar ? Could you please give me a. hint as to why do you need to divide the loss by this number? and if when you were testing, the loss was high as well ? 

Your input would be very much appreciated.

**Comment by orybkin** [comment]
Hi,

I am not sure whether there is an actual issue here. The loss being "very high" is not an issue, unless you are not able to replicate the results from the paper. You could compare table 2 or table 3 for MNIST. Let me know if you are not able to replicate the results using my code.

Regarding dividing the loss, I vaguely remember that the VAE was working a bit better when dividing the loss (in this case by about 2000). Adam is supposed to be scale-invariant, but it could be that high loss values yield numerical issues, or mess up with Adam optimization before it can even estimate the normalization constant reliably.

**Comment by orybkin** [comment]
Closing for now unless there is a replication issue

**Comment by orybkin** [comment]
Closing for now unless there is a replication issue

**Comment by scigeek72** [comment]
I know it is closed now but wanted to thank you for your time. I didn't try to use my code on MNIST or other image sets. I will try to do so and let you know if I face any issue.
